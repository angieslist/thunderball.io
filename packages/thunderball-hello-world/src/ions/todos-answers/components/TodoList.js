import React from 'react';
import PropTypes from 'prop-types';
import Todo from './Todo';

const TodoList = ({ todos, toggleTodo }) => (
  <div>
    <If condition={!todos.length}>
      <p>
        No TODOs
      </p>
    </If>
    {todos.map(todo =>
      (<Todo
        key={todo.id}
        {...todo}
        onClick={() => toggleTodo(todo.id)}
      />),
    )}
  </div>
);

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    completed: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
  }).isRequired).isRequired,
  toggleTodo: PropTypes.func.isRequired,
};

export default TodoList;
