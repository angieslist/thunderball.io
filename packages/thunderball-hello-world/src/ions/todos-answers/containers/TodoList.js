import { connect } from 'react-redux';
import { toggleTodo } from '../actions';
import TodoList from '../components/TodoList';

const mapStateToProps = state => ({
  todos: state.todos,
});

const mapDispatchToProps = {
  toggleTodo,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TodoList);
