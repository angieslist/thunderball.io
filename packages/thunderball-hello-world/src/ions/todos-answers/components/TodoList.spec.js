import React from 'react';
import { shallow } from 'enzyme';

import TodoList from './TodoList';

const setup = (setupProps = {}) => {
  const defaultProps = {
    todos: [],
    toggleTodo: jest.fn(),
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <TodoList {...props} />,
  );

  return {
    props,
    wrapper,
  };
};

describe('TodoList', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('with todos', () => {
    const { wrapper, props } = setup({
      todos: [
        {
          text: 'Test AddTodo',
          completed: false,
          id: 0,
        },
      ],
    });

    test('renders a list of todos', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('todo onClick calls toggleTodo', () => {
      wrapper.find('Todo').simulate('click');
      expect(props.toggleTodo).toHaveBeenCalled();
    });
  });
});
