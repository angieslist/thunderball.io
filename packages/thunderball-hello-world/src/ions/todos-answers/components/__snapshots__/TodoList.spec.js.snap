// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders without crashing 1`] = `
<div>
  <p>
    No TODOs
  </p>
</div>
`;

exports[`TodoList with todos renders a list of todos 1`] = `
<div>
  <Todo
    completed={false}
    id={0}
    key="0"
    onClick={[Function]}
    text="Test AddTodo"
  />
</div>
`;
